### PR TITLE
Issue 21313 multitree do not keep tree order 2

### LINKED ISSUE #21313: Push publishing sometimes changes the order of contentlets in the multi-tree table on multilingual pages
**Describe the bug**
Push publishing changes the order of contentlets in the multi-tree table on multilingual pages where there are unique contentlets only present on some pages. 

Related Ticket: https://dotcms.zendesk.com/agent/tickets/105842
Reproduced on: Master, 21.06.3

**To Reproduce**
1. From a demo starter, create a page in English
2. Add English only or English/Spanish contentlets to it
3. Create a Spanish version of the page
4. Add Spanish only contentlets to the page
5. Check the multi_tree table for that page

`select * from multi_tree where parent1 = '{PAGE_IDENTIFIER}' order by tree_order ASC;`

6. You will see there are duplicated order numbers that represent individual contents on the separate languages. 
7. Push publish this page in all languages to a receiver.
8. Check the multi_tree table on the receiver. The ordering will be changed to be sequential and the content will not appear in the correct order. 

**Expected behavior**
The order numbers should be maintained on push.

**Additional context**
This may cause other unexpected behaviour and should likely be changeable via config var that defaults to the current behaviour.

### PR DESCRIPTION
**Pull request by jdotcms**
Most of the case the multi tree already has the tree order set, so we do not want to override it. Only the edit mode should

### CODE REVIEW

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void setMultiTree_on_diff_orders_should_not_reorder() throws IOException, DotSecurityException, DotDataException {
+
+        final String pageIdentifier = "e65ce6c5-6b03-45f8-b2a8-90ee80d17987";
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtilTest.java:32]: Does this page always exist? Why don't you create a new page?

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void setMultiTree_on_diff_orders_should_not_reorder() throws IOException, DotSecurityException, DotDataException {
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtilTest.java:30]: I would add a comment on this test to understand with more details the scenario being tested and the expected result

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtilTest.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtilTest.java:32]: does not matter b/c the multi tree does not have any integrity

**Review by wezell — APPROVED** [review]
Changes look good

**Review by erickgonzalez — CHANGES_REQUESTED** [review]
Tests didn't run b/c compiling issue:
```
/build/src/core/dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtilTest.java:47: error: cannot find symbol
        HandlerUtil.setMultiTree(pageIdentifier, pageInode, pageLanguage, wrapperMultiTree, modUser);
        ^
  symbol:   variable HandlerUtil
  location: class HandlerUtilTest
  ```

**Comment by erickgonzalez** [comment]
The issue with EE jar was fixed after adding the HandlerUtil to the proguard.

**Pull request merged**